### PR TITLE
Add additional info to AcquireFailure exception

### LINKED ISSUE #35: When raising AcquireFailure provide more info for debugging
For example, the name of the lock and why it's failed to acquire.

### PR DESCRIPTION
**Pull request by bchopson**
Fixes GH-35

### CODE REVIEW

File: pals/core.py [hunk 1]
```diff
@@ -112,11 +119,20 @@ def acquire(self, blocking=None, acquire_timeout=None):
...
+                return True
+            else:
+                raise AcquireFailure(self.name, 'result was: {retval}')
         except sa.exc.OperationalError as e:
             if 'lock timeout' not in str(e):
                 raise
             log.debug('Lock acquire failed due to timeout')
+            raise AcquireFailure(self.name, 'Failed due to timeout')
+
+    def acquire(self, blocking=None, acquire_timeout=None) -> bool:
```
Comment (bchopson) [inline pals/core.py:132]: Preserves existing API (return False if lock fails to acquire)

File: pals/core.py [hunk 2]
```diff
@@ -76,21 +81,23 @@ def _lock_num(self, name):
...
         self.engine = engine
         self.conn = None
         self.lock_num = lock_num
+        self.name = name
         self.blocking = blocking
         self.acquire_timeout = acquire_timeout
         self.shared_suffix = '_shared' if shared else ''
 
-    def acquire(self, blocking=None, acquire_timeout=None):
+    def _acquire(self, blocking=None, acquire_timeout=None) -> bool:
```
Comment (bchopson) [inline pals/core.py:100]: raises AcquireFailure if lock cannot be acquired

File: docker-compose.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 version: '2'
 services:
   pals-pg:
-      image: postgres:10-alpine
+      image: postgres:15-alpine
       container_name: pals-pg
       ports:
         - '127.0.0.1:54321:5432'
```
Comment (guruofgentoo) [inline docker-compose.yaml]: We should change this to support env file overrides ala https://github.com/level12/keg/blob/master/docker-compose.yaml#L20 (needs the 2.1 version above)

File: .gitignore [hunk 4]
```diff
@@ -11,3 +11,4 @@ build
 .pytest_cache
 .vscode
 .env
+docker-compose.override.yaml
```
Comment (guruofgentoo) [inline .gitignore]: Fine to keep this, I suppose. But mostly we should be using env file overrides with vars in the compose script, so we get those file updates if they change.

**Review by guruofgentoo — CHANGES_REQUESTED** [review]
Notes left on the docker compose arrangement. I don't see anything backwards-incompatible for the acquiring update itself, but you can run something with it to verify that. Once that's confirmed and the other notes addressed, we can merge/release.

**Comment by bchopson** [comment]
@guruofgentoo I have the test suite passing on an internal project which uses PALS, when pinned to this commit. I think this PR is good to go.

**Comment by guruofgentoo** [comment]
@bchopson pals 0.3.3 has been released with this update.

**Pull request merged**